tests/CMakeLists.txt

```diff
@@ -107,12 +107,10 @@ add_executable(sdbus-c++-unit-tests ${UNITTESTS_SRCS})
 target_compile_definitions(sdbus-c++-unit-tests PRIVATE
     LIBSYSTEMD_VERSION=${LIBSYSTEMD_VERSION}
     SDBUS_HEADER=<${LIBSYSTEMD_IMPL}/sd-bus.h>)
-target_compile_features(sdbus-c++-unit-tests PUBLIC cxx_std_20)
```

Comment:
This way you can never test any c++20 dependent code again.

What if we use [`try_compiles`](https://cmake.org/cmake/help/latest/command/try_compile.html) to check if c++20 is supported and if yes, we add a second target with just `std=c++20` enabled? So we have 4 tests instead of 2? (or 3, if we leave out the intregation tests)

Reply:
Nevermind, I brought the cxx20 lines back, and I will leave them out in `release/v2.0` when rebasing.